tensorflow/contrib/tensorrt/python/trt_convert.py

```diff
@@ -46,11 +46,13 @@
 from tensorflow.python.training import saver
 
 if _six.PY2:
-  _to_bytes = lambda s: s
-  _to_string = lambda s: s
+  _to_bytes = lambda s: s.encode("utf-8", errors="surrogateescape") \
```

Comment:
Smit, Sorry for the delay, came back from holidays :)

For checking the types, we can replace `if isinstance(s, str)` by `if isinstance(s, six.text_type)` in py3 for example.

But for doing the actual conversion (encode and decode), I tried to use `six.b` and `six.u` but no success. These two functions seem to be a no-op in some cases.

Reply:
"it's not clear to me why we are converting to bytes before adding values to protocol buffer."
I think we always convert to binary before storing in protobuf and convert to string after reading from protobug (if we need to act on a string). Isn't this the case?

Reply:
Happy New Year!

I was thinking of following to avoid making use of _six.PY2 and _six.PY3.

_to_bytes = lambda s: s.encode("utf-8", errors="surrogateescape") if isinstance(s, _six.text_type) else s
_to_string = lambda s: s.decode("utf-8") if isinstance(s, _six.binary_type) else s

Waiting for Lambda to take a look at this PR and also confirm the requirement of converting to bytes before adding values to protocol buffer. I haven't yet investigated about that.

Reply:
For details see [here](https://developers.google.com/protocol-buffers/docs/proto3#scalar), which I think is confusing, so I think it's safe to always do the str/bytes conversion.

Reply:
Can we use _six.text_type and _six.binary_type to simplify the functions? For example,

def _to_bytes(s):
  """Returns encoded of s if s is a sequence of chars otherwise returns s."""
  if isinstance(s, _six.text_type):
    return s.encode("utf-8", errors="surrogateescape")
  return s

Reply:
@smit-hinsu Thanks for the review. I made the changes.